### PR TITLE
Message remove

### PR DESCRIPTION
**Pull request by AlexZavkn**
screenshot_18
screenshot_17

### CODE REVIEW

File: features/page/homePage.js [hunk 1]
```diff
@@ -16,6 +14,19 @@ const messagesRefSelector = 'td.xY.bnk.byv';
...
 const messagesSelector = 'div[id=":23"]';
+const messagesCheckboxesSelector = 'td.oZ-x3.xY';
+const messageActionCheckboxSelector = '[role="checkbox"]';
+const messagesRemovalButtonSelector = 'div.T-I.J-J5-Ji.nX.T-I-ax7.T-I-Js-Gs.ar7';
+const totalAmountOfMessagesHeader = 'div.J-J5-Ji.amH.J-JN-I';
+const amountOfMessagesNumberSelector = 'span.ts';
+const parentOftextPreviewElementSelector = 'td.xY.a4W';
+const textOfPreviewedMessageSelector = 'span.y2';
+const moreOptionSelector = '.J-Ke.n4.ah9';
+const trashButtonSelector = '[href="https://mail.google.com/mail/u/0/#trash"]';
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: can we  make selector more easy?

File: features/page/homePage.js [hunk 2]
```diff
@@ -26,15 +37,31 @@ const homePage = Object.create({
     browser.waitForVisible(homePageSelector);
   },
 
-  headerMessagesAmountValue: function headerMessagesAmountValue() {
-    return Number(browser.element(messagesHeaderSelector)
-      .element(totalMessagesHeaderSelector).getText());
+  headerMessagesOnPageAmountValue: function headerMessagesAmountValue() {
+    this.waitForVisible();
+    const parentElementId = browser.element(totalAmountOfMessagesHeader).value.ELEMENT;
+    const lastNumberOfMessageOnPage = Number(browser
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: we can get only 1 request

File: features/page/homePage.js [hunk 3]
```diff
@@ -26,15 +37,31 @@ const homePage = Object.create({
...
+      .elementIdElements(parentElementId, amountOfMessagesNumberSelector)
+      .value[1].getText());
+    const firstNumberOfMessageOnPage = Number(browser
+      .elementIdElements(parentElementId, amountOfMessagesNumberSelector)
+      .value[0].getText());
+    return lastNumberOfMessageOnPage - (firstNumberOfMessageOnPage + 1);
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: why +1?

File: features/page/homePage.js [hunk 4]
```diff
@@ -26,15 +37,31 @@ const homePage = Object.create({
...
   },
 
   messagesOnPageValue: function messagesOnPageValue() {
+    this.waitForVisible();
     return browser.elements(messageCommonSelector).value.length;
   },
 
+  messagesOverallValue: function messagesOverallValue() {
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please change on more accurate

File: features/page/homePage.js [hunk 5]
```diff
@@ -26,15 +37,31 @@ const homePage = Object.create({
...
+    this.waitForVisible();
+    const parentElementId = browser.element(totalAmountOfMessagesHeader).value.ELEMENT;
+    return Number(browser
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please create const for it

File: features/page/homePage.js [hunk 6]
```diff
@@ -59,6 +86,14 @@ const homePage = Object.create({
     return unreadMessagesAmount;
   },
 
+  selectLastMessage: function selectLastMessage() {
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: if you click on top message, please rename 'clickOnTopEmail'

File: features/page/homePage.js [hunk 7]
```diff
@@ -96,6 +131,38 @@ const homePage = Object.create({
     lastMessageRef.waitForVisible();
     lastMessageRef.click();
   },
+
+  getLastMessageText: function getLastMessageText() {
+    this.waitForVisible();
+    const parentElementId = browser.element(parentOftextPreviewElementSelector).value.ELEMENT;
+    textOfMessagePreview = browser
+      .elementIdElements(parentElementId, textOfPreviewedMessageSelector)
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: why not only `elementIdElement `?

File: features/page/homePage.js [hunk 8]
```diff
@@ -96,6 +131,38 @@ const homePage = Object.create({
     lastMessageRef.waitForVisible();
     lastMessageRef.click();
   },
+
+  getLastMessageText: function getLastMessageText() {
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please rename `getTextFromTopEmail`

File: features/page/homePage.js [hunk 9]
```diff
@@ -16,6 +14,19 @@ const messagesRefSelector = 'td.xY.bnk.byv';
 const tableRowSelector = 'tr';
 const nextPageButtonSelector = 'div[id=":ha"]';
 const messagesSelector = 'div[id=":23"]';
+const messagesCheckboxesSelector = 'td.oZ-x3.xY';
+const messageActionCheckboxSelector = '[role="checkbox"]';
+const messagesRemovalButtonSelector = 'div.T-I.J-J5-Ji.nX.T-I-ax7.T-I-Js-Gs.ar7';
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: may be rename `removeMessageButton`?

File: features/page/homePage.js [hunk 10]
```diff
@@ -96,6 +131,38 @@ const homePage = Object.create({
...
+  getLastMessageText: function getLastMessageText() {
+    this.waitForVisible();
+    const parentElementId = browser.element(parentOftextPreviewElementSelector).value.ELEMENT;
+    textOfMessagePreview = browser
+      .elementIdElements(parentElementId, textOfPreviewedMessageSelector)
+      .value[0].getText();
+    return textOfMessagePreview;
+  },
+
+  removeMessagesSelected: function removeMessagesSelected() {
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: may be rename `clickRemoveMessageButton`?

File: features/page/homePage.js [hunk 11]
```diff
@@ -96,6 +131,38 @@ const homePage = Object.create({
...
+    messagesAmount = this.messagesOverallValue();
+    this.getLastMessageText();
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: what for it?

File: features/page/homePage.js [hunk 12]
```diff
@@ -96,6 +131,38 @@ const homePage = Object.create({
...
+    browser.waitForVisible(messagesRemovalButtonSelector);
+    browser.click(messagesRemovalButtonSelector);
+  },
+
+  getMessagesAmountWrittenPreviously: function removeMessagesSelected() {
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please fix copypast issue `removeMessagesSelected`

File: features/step_definitions/home.js [hunk 13]
```diff
@@ -18,8 +18,21 @@ module.exports = function home() {
     homePage.clickSendMessageButton();
   });
 
+  this.When(/^I remove last message in this box$/, function removingLastMessage() {
```
Comment (AleksandrHorev) [inline features/step_definitions/home.js]: please reaname `^I remove email from top in this box`

File: features/step_definitions/home.js [hunk 14]
```diff
@@ -18,8 +18,21 @@ module.exports = function home() {
...
   this.Then(/^I am getting "([^"]*)" message at this box$/, function expectingMessage(textOfInspectingMessage) {
     homePage.openLastMessage();
     expect(messagePage.textOfOpenedMessage()).toBe(textOfInspectingMessage);
   });
+
+  this.Then(/^I am getting this message deleted$/, function countingDeleted() {
+    expect(homePage.messagesOverallValue() + 1).toBe(homePage.getMessagesAmountWrittenPreviously());
+  });
+
+  this.Then(/^I am getting message in trash$/, function trashInspection() {
```
Comment (AleksandrHorev) [inline features/step_definitions/home.js]: please remove` I see email from previous step  in trash`

File: features/step_definitions/home.js [hunk 15]
```diff
@@ -18,8 +18,21 @@ module.exports = function home() {
...
+    homePage.selectLastMessage();
+    homePage.removeMessagesSelected();
+  });
+
   this.Then(/^I am getting "([^"]*)" message at this box$/, function expectingMessage(textOfInspectingMessage) {
     homePage.openLastMessage();
     expect(messagePage.textOfOpenedMessage()).toBe(textOfInspectingMessage);
   });
+
+  this.Then(/^I am getting this message deleted$/, function countingDeleted() {
```
Comment (AleksandrHorev) [inline features/step_definitions/home.js]: I didn't understand this step..may be rename like `I see count emails reduce on "" on home page`

File: features/page/homePage.js (hunk 11 above)
Comment (AlexZavkn) [reply features/page/homePage.js]: It`s hand-made save of "global state" (that would be moved to the world.js soon)

File: features/page/homePage.js (hunk 3 above)
Comment (AlexZavkn) [reply features/page/homePage.js]: image

**Comment by AlexZavkn** [comment]
screenshot_20
screenshot_21

**Pull request merged**